.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "test/testdata/golden/googleapis"]
```

Comment:
Okay, I just looked back and saw:

> One thing that might be circleci specific is that the git submodule folder won't be available in the artman container running in circleci. To make it work, I add a git clone step in the golden-test circleci config, which is even better than a locked submodule folder, because that helps capture the potential regression caused by upstream common proto/config changes.

First of all, that is not CircleCI specific. `git clone` does not clone submodules by default. You have to provide `--recurse-submodules` or use `git submodule update` later. (This is an _extremely_ common misunderstanding of how submodules work.)

More importantly, you do not want a submodule. :-) You want to track HEAD. Kill the submodule and check out to a separate location.

The lousy part of this is that it is going to make it somewhat ornery to run the test locally. My recommendation would be to set it up with the following logic:

  1. If `CIRCLECI` is set, check `googleapis` out to a fixed location.
  2. Else, if `GOOGLEAPIS_DIR` is set, skip checkout entirely and just use whatever is there.
  3. Else, `pytest.skip`.